docs/updater.md

```diff
@@ -0,0 +1,102 @@
+  * `std::function<void()>`: For general use.
+  * `std::function<void(const avk::graphics_pipeline&)>`: In combination with `gvk::destroying_graphics_pipeline_event`.
+  * `std::function<void(const avk::compute_pipeline&)>`: In combination with `gvk::destroying_compute_pipeline_event`.
+  * `std::function<void(const avk::ray_tracing_pipeline&)>`: In combination with `gvk::destroying_ray_tracing_pipeline_event`.
+  * `std::function<void(const avk::image&)>`: In combination with `gvk::destroying_image_event`.
+  * `std::function<void(const avk::image_view&)>`: In combination with `gvk::destroying_image_view_event`.
+* `then_on(event... events)`: bind the following up `update` or `invoke` call to a list of events. However, unlike `on(event... events)`, the given events will be evaluated _after_ the updates induced by previous `update` or `invoke` calls have finished execution.
+
+It is important to note that the order in which the events are evaluated is not necessarily determined by the order of the events in each separate list.
+
+### Examples
+
+```
+mPipeline.enable_shared_ownership(); // Make it usable with the updater
+mUpdater->on(gvk::swapchain_resized_event(gvk::context().main_window()), 
+      gvk::shader_files_changed_event(mPipeline))
+  .update(mPipeline);
+```
+
+In the above example, the `gvk::updater mUpdater` is configured so that the `mPipeline` is updated when at least one of two _events_ occurs: either when the swapchain of the main window has been resized, or when one of the `mPipeline`'s shader files have been changed on the file system. 
+
+```
+mUpdater->on(
+      gvk::swapchain_format_changed_event(gvk::context().main_window()),
+      gvk::swapchain_additional_attachments_changed_event(gvk::context().main_window())
+   ).invoke([this]() {
+      // update render pass here
+   }).then_on(
+      gvk::swapchain_changed_event(gvk::context().main_window()),
+      gvk::shader_files_changed_event(mPipeline)
+   ).update(mPipeline);
+```
+
+In the above example, the renderpass belonging to `mPipeline` requires recreation only if the events `swapchain_format_changed_event` or `swapchain_format_changed_event` arise. Then it is evaluated whether _any type_ of swapchain recreation has occured and if so, subsequently the pipeline is updated. It is important to note here: The evaluation of events given to `then_on()` is only temporaly restricted by the previous list: this evaluation will occur in any case.
+
+## Common Use-Cases
+* Swapchain recreation can lead to changes to image properties. This may cause any pipeline, renderpass, framebuffer or image view that directly works with swapchain to end in an invalid state. The updater mechanism is meant to streamline the recreation process of those objects, as well as to allow any user-defined behaviour to execute in cases where such events occur.
```

Comment:
```suggestion
* Swapchain recreation can lead to changes in image properties. This may cause any pipeline, renderpass, framebuffer or image view that directly works with swapchain to end in an invalid state. The updater mechanism is meant to streamline the recreation process of those objects, as well as to allow any user-defined behaviour to execute in cases where such events occur.
```

Hmm.. not sure about the above. Would "Swapchain recreation can require changes in other, dependent images" be more appropriate?